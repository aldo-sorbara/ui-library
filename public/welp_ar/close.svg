<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
