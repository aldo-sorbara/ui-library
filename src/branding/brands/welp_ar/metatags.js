export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
