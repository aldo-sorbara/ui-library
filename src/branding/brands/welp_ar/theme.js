import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
