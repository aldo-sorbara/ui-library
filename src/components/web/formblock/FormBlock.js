import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

/**
 * @render react
 * @name FormBlock
 * @description FormBlock Component for Wenance
 */

const FormBlock = props => {
  const { children, ...rest } = props;
  const Component = StyledBlock;
  return <Component {...rest}>{children}</Component>;
};

const StyledBlock = styled.div`
  padding-bottom: ${props => props.theme.spacer.xLarge};
`;

StyledBlock.propTypes = {
  children: PropTypes.node.isRequired
};

StyledBlock.defaultProps = {
  theme: {
    spacer: {
      xLarge: '2rem'
    }
  }
};

export default FormBlock;
