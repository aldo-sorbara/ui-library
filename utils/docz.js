export const styles = {
  display: 'flex',
  flex: 1,
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'space-around'
};
