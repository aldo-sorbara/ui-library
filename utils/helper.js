import { shallow, mount } from 'enzyme';

import { ThemeConsumer } from 'styled-components';

export const shallowWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

export const mountWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return mount(children);
};
