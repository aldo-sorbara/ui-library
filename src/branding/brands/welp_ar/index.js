import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
