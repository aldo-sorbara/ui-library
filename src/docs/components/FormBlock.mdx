---
name: FormBlock
menu: Components
route: /formblock
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import FormBlock from '../../components/web/formblock';
import Input from '../../components/web/input';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';
import { styles } from '../../../utils/docz';

# FormBlock

<PropsTable of={FormBlock} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <FormBlock>
        <Input
          label='Ingrese su celular'
          name='validationcellphone'
          descriptionBottom
          type='number'
          prefix='15'
          description='Sin el 0 y 15'
          placeholder='42424242'
          data='ingrese-celular'
        />
      </FormBlock>
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/formblock--with-default-props)
