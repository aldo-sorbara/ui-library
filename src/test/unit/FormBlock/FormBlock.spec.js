import React from 'react';
import FormBlock from '../../../components/web/formblock';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('FormBlock component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <FormBlock className='test-formBlock'>FormBlock container</FormBlock>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render a formBlock with padding', () => {
    const wrapper = mountWithTheme(
      <FormBlock className='test-formBlock'>FormBlock</FormBlock>,
      welp_ar.theme
    );
    expect(wrapper).toHaveStyleRule('padding-bottom', '2rem');
  });
});
