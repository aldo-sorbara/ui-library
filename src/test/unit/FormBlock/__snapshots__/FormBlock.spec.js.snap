// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBlock component should render snapshot 1`] = `
.c0 {
  padding-bottom: 2rem;
}

<div
  className="test-formBlock c0"
>
  FormBlock container
</div>
`;
